src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -46,8 +46,12 @@ def create_from_spec_one(host: str, drive_selection: DriveSelection) -> Optional
             if self.mgr.inventory.has_label(host, '_no_schedule'):
                 return None
 
+            # Note: OSD id claims in CRUSH map are linked to the bare name of
+            # the hostname. Use bare name in case of FQDN hostnames.
+            osd_id_claims_for_host = osd_id_claims.get(host.split(".")[0], [])
```

Comment:
I somewhat don't really like the fact that the data structure `osd_id_claims` contains keys that are not consumable by the rest of the module. Can re move the responsibility from the consumer to the creator of `osd_id_claims`? 

```python
class OsdIdClaims:
    def __init__(self, d: dict):
        self._d = d
    def for_host(host):
        return self._d[host.split(".")[0]]
```

that would avoid code duplication and make sure that no one can get this wrong.